include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -1036,9 +1039,9 @@ TridiagResult<T, Device::CPU> BandToTridiag<Backend::MC, D, T>::call_L(
     return {std::move(mat_trid), std::move(mat_v)};
   }
 
-  auto comm = ex::just(grid.fullCommunicator().clone());
+  auto mpi_chain = grid.full_communicator_pipeline();
   // Need a pipeline of comm for broadcasts.
-  common::Pipeline<comm::Communicator> comm_bcast(grid.fullCommunicator().clone());
+  auto mpi_chain_bcast = grid.full_communicator_pipeline();
```

Comment:
I added a comment in https://github.com/eth-cscs/DLA-Future/pull/993/commits/47adaa0e66a303256248e444b346fe1b65d93aa3. Could you please check if it explains the situation clearly enough for you?

Reply:
Someone might ask why you need 2 pipelines...
I would add in someway in the comment, 1 works 2 it's better...

Reply:
Good point, I'll add that as well!

Reply:
A summary from your two comments (feel free to adjust it)

> The algorithm would still work if we only defined one pipeline below and used that for both p2p communication and broadcasts (since mpi_chain is used fully before mpi_chain_bcast, i.e. not interleaved).
> However, using two pipelines is beneficial since it allows the two different types of communications to be submitted concurrently. This is true in case there are actually two resources (pipelines.size() = 2), otherwise it will fallback to the "single" pipeline case anyway (ending up serialising the calls).

Reply:
I'm ok with the suggestion from @albestro as well, or the one that's there now. @rasolca do you have a preference?

Reply:
Ok with @albestro suggestion.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/993/commits/e16a63b102f6cd0f849b6c18bfe78a3ca916a757